drv/lpc55-swd/src/main.rs

```diff
@@ -550,7 +571,10 @@ impl idl::InOrderSpCtrlImpl for ServerImpl {
         let r =
             Reg::from_u16(register).ok_or(SpCtrlError::InvalidCoreRegister)?;
         self.write_single_target_addr(DCRSR, r as u32)
-            .map_err(|_| SpCtrlError::Fault)?;
+            .map_err(|e| {
+                let err: SpCtrlError = e.into();
+                err
+            })?;
```

Comment:
nitpick, take it or leave it: could also be 
```suggestion
            .map_err(|e| SpCtrlError::from(e))?;
```

Reply:
Heh I think this is from an earlier pass where the compiler was refusing to do the conversion. I'll clean it up.

Reply:
it's also possible that i am just wrong! :)